numpy/core/src/multiarray/dragon4.c

```diff
@@ -2714,7 +2714,7 @@ Dragon4_PrintFloat_Intel_extended128(
  * becomes more common.
  */
 static npy_uint32
-Dragon4_PrintFloat_IEEE_binary128(
+Dragon4_PrintFloat_IEEE_binary128_le(
```

Comment:
This function is for quad precision floats, I doubt that ARM implements that.

Reply:
Without this change, I got warnings for implicit definition of `Dragon4_PrintFloat_IEEE_binary128_le` and unused symbol for `Dragon4_PrintFloat_IEEE_binary128`

Additionally, it failed on import looking for `Dragon4_PrintFloat_IEEE_binary128_le`

It is very possible that the root cause is actually the identification of what representation it should be using. I've been looking at https://github.com/numpy/numpy/blob/20a80e08a9031cc6c69b168a0988443d15b8101f/numpy/core/setup_common.py#L343 to see if there's something incorrect there.

I've created the object file outside of running setup_common, and confirmed that the sequence that identifies IEEE_QUAD_LE is indeed present in the `od -b` dump

Reply:
Given the error that was thrown, I tried the naive approach of just editing the function name, not fully expecting it to work, but it did (at least as far as building was concerned, still have some test failures)

Reply:
Hmm, LLVM does seem to have software support for quad precision, so I think we need to track this down. What is your compiler tool change. I'm concerned about the endian part. @ahaldane Comment?

Reply:
The use of quad precision seems to depend on a compiler flag, `-mfloat128`. I don't know what the default value is.

Reply:
`-mfloat128` does not seem to be a valid option on aarch64 but is available for ppc64le

Reply:
```bash
$ gcc --version
gcc (Debian 7.3.0-24) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE
```

```bash
$ uname -a
Linux localhost 3.18.0-17554-g9194949d4df2 #1 SMP PREEMPT Fri Jun 22 17:45:52 PDT 2018 aarch64 GNU/Linux
```

My machine is a little endian aarch64 chromebook

Reply:
Maybe wrong flag, but it is there somewhere, https://llvm.org/docs/LangRef.html#floating-point-types. We should be detecting on the byte representation, so either there is an error in that or the compiler is actually producing quad floats.

Reply:
OK we need to have two functions, one with the `_be` extension and one with the `_le` extension. This function, despite the comment, looks to be big endian with the little endian version in #11570. The difference is only four lines, so it would be nice to have wrappers for the endian specific versions that called into the common code, but the easy fix would be copy the function from # 11570 here with the appropriate name.

Reply:
See similar wrappers for the various extended precision alignments.

Reply:
I think all the wrappers need to do is swap the a and b members of buf128, so maybe only one wrapper to do that for the `_le` version and name this function with the `_be` extension.

Reply:
I've pushed the easy fix, cherry-picking the commit from my other PR, I'll work on the wrapper func now